pkg/storegateway/indexheader/binary_reader.go

```diff
@@ -884,13 +923,27 @@ func (r *BinaryReader) LabelValues(name string, filter func(string) bool) ([]str
 		} else {
 			d.Skip(skip)
 		}
-		s := yoloString(d.UvarintBytes()) // Label value.
-		if filter == nil || filter(s) {
-			values = append(values, s)
+
+		value := yoloString(d.UvarintBytes())
```

Comment:
I would also rename this to `unsafeValue` (even it this code will probably be removed soon).